docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+## III. Syntax and Grammar
+
+The syntax and grammar of Cadence define the formal structure of programs written in the language. This section outlines the key elements that make up Cadence programs, including how contracts, interfaces, functions, variables, and statements are declared and used, as well as the rules governing expressions and control flow.
+
+---
+
+### 1. Program Structure
+
+A Cadence program is composed of a combination of **contracts**, **interfaces**, **transactions**, and **scripts**. These building blocks allow developers to create decentralized applications that interact with the Flow blockchain.
+
+- **Contracts**: Define the structure and behavior of reusable modules that manage state and execute business logic on the blockchain.
+- **Interfaces**: Declare abstract behavior that contracts or other types can implement, supporting both static and dynamic contracts.
+- **Transactions**: Define the operations that modify the blockchain state and handle interactions between accounts.
+- **Scripts**: Read-only code used to query data from the blockchain without modifying its state.
+
+Each Cadence program must specify its purpose within the structure defined above, allowing for secure interactions with the blockchain.
+
+- **EBNF**:
+  ```ebnf
+  program
+    : ( declaration ';'? )* EOF
+    ;
+  ```
+
+---
+
+### 2. Declarations
+
+In Cadence, declarations form the backbone of a program, enabling the definition of data structures, resources, contracts, functions, variables, events, and other key components. A declaration defines the type and scope of these components, laying out how they interact with each other within the program. Cadence’s declaration model is designed to be clear and explicit, ensuring that every element in a program is well-typed, properly scoped, and initialized according to strict language rules.
+
+Cadence is a resource-oriented programming language, and the declaration system reflects its focus on security and ownership. Resources, which are unique and must be used exactly once, are declared and managed alongside more traditional constructs like structs and functions. This approach ensures that programs written in Cadence follow safe patterns for managing digital assets, preventing duplication and loss while supporting complex, scalable applications.
+
+The Declarations section introduces the grammar and syntax for various types of declarations in Cadence. From basic data structures such as structs and variables, to more complex entities like contracts and entitlement mappings, this section covers the formal rules for creating and using these elements within the language. Each declaration type is defined using Cadence’s specific Extended Backus-Naur Form (EBNF), providing a precise specification for developers and implementers to follow.
```

Comment:
Maybe use a "kinds of declarations" / "declaration kind" instead of "types of declarations", because "type" might be misunderstood to mean _type_. See also the implementation, where this naming is used to avoid ambiguity/mistakes

Reply:
I agree. This has been updated.